script.skin.helper.skinbackup/resources/lib/backuprestore.py

```diff
@@ -0,0 +1,443 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+'''
+    script.skin.helper.skinbackup
+    Kodi addon to backup skin settings
+'''
+
+import xbmc
+import xbmcvfs
+import xbmcgui
+import xbmcaddon
+from utils import log_msg, ADDON_ID, SKIN_NAME
+from utils import recursive_delete_dir, get_clean_image, normalize_string
+from utils import zip_tofile, unzip_fromfile
+from dialogselect import DialogSelect
+from xml.dom.minidom import parse
+from datetime import datetime
+import os
+
+
+class BackupRestore:
+    '''Main BackupRestore class providing methods to backup and restore skin settings'''
+    params = {}
+
+    def __init__(self):
+        '''Initialization and main code run'''
+        self.addon = xbmcaddon.Addon(ADDON_ID)
+
+    def __del__(self):
+        '''Cleanup Kodi Cpython instances on exit'''
+        del self.addon
+
+    def backup(self, filters=None, backup_file="", silent=False):
+        '''create skin backup'''
+        if not filters:
+            filters = []
+
+        if not backup_file:
+            return
+
+        # create temp path
+        temp_path = self.create_temp()
+
+        # backup skinshortcuts preferences
+        if not filters or (filters and "skinshortcuts" in filters):
+            self.backup_skinshortcuts(temp_path + "skinshortcuts/")
+
+        # backup skin settings
+        if "skinshortcutsonly" not in filters:
+            skinsettings_path = os.path.join(temp_path, u"guisettings.txt")
+            self.backup_skinsettings(skinsettings_path, filters, temp_path)
+
+        # zip the backup
+        zip_temp = u'special://temp/skinbackup-%s.zip' % datetime.now().strftime('%Y-%m-%d %H.%M')
```

Comment:
please do not use the kodi temp folder.

addons need to use their own addon_data subfolder for data storage.